validate/validate.go

```diff
@@ -95,6 +100,53 @@ func NewPriorityCheck() InRangeIncludingCheck[int64] {
 	return newInRangeExcludingCheck[int64]("%v priority value %v cannot be %v than %v", 100, 4096)
 }
 
+type maxLengthValidator struct {
+	maxLength int
+}
+
+func (m maxLengthValidator) Check(value string, valueType interface{}) error {
+	if len(value) < 1 {
+		return fmt.Errorf("%v cannot be empty", valueType)
+	} else if len(value) > m.maxLength {
+		return fmt.Errorf("%v maximum length is %v", valueType, m.maxLength)
+	}
+	return nil
+}
+
+func NewDbUsernameValidator(engine resourcespb.DatabaseEngine, version string) Validator {
+	switch engine {
+	case resourcespb.DatabaseEngine_MYSQL:
+		switch version {
+		case "5.6":
+			return &maxLengthValidator{16}
+		default:
+			return &maxLengthValidator{32}
+		}
+	case resourcespb.DatabaseEngine_POSTGRES:
+		return &maxLengthValidator{63}
+	case resourcespb.DatabaseEngine_MARIADB:
+		return &maxLengthValidator{80}
+	default:
+		return nil
```

Comment:
Right now, if you set database engine to unknown, this will panic with a null pointer.

This should either:
1. Return a dummy validator that never returns an error
2. Return an error that is then caught by the caller, and then makes sure .Check() is not called.

I prefer the first one but I'll leave it it at your discretion

Reply:
fixed with dummyValidator